### PR TITLE
Sad path: User can delete a favorite from a playlist with /playlists/:id/favorites/:id endpoint

### LINKED ISSUE #54: User can delete a specific favorite from a playslist with /playlists/:id/favorites/:id endpoint
User can delete a specific favorite from playlist by sending playlist id and favorite id in params to
`DELETE /api/v1/playlists/:id/favorites/:id`

User can expect `404 status` if playlist ID or favorite ID cannot be found

Acceptance criteria:

- [x]  All tests passing
- [x]  Sad path
  - [x] `404 status`
- [ ]  Peer review

### PR DESCRIPTION
**Pull request by lrs8810**
## Type of change 

- [x] :beetle: Bug fix (non-breaking change which fixes an issue)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 
- Add 404 sad path spec for `DELETE /playlists/:id/favorites/:id `
- Add implementation of 404 sad path for `DELETE /playlists/:id/favorites/:id` route

## How Has This Been Tested?

- [x] :black_square_button: Integration testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :eight_spoked_asterisk: I have commented on my code, particularly in hard-to-understand areas
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: routes/api/v1/playlists.js [hunk 1]
```diff
@@ -119,13 +119,34 @@ router.get('/:playlistId/favorites', (req, res) => {
...
   let favoriteId = request.params.favoriteId
-  database('playlists_favorites').where({playlist_id: playlistId, favorite_id: favoriteId}).del()
-  .then(favorite => {
-    if (favorite > 0) {
-      response.status(204).send();
-    }
+
+  database('playlists_favorites').where({playlist_id: playlistId})
+  .then(playlists => {
+    if (playlists.length === 0) {
```
Comment (lrs8810) [inline routes/api/v1/playlists.js]: If you have any feedback on how I can make these conditionals/validation checks better, I am happy to make the changes.

File: tests/playlistFavorites.spec.js [hunk 2]
```diff
@@ -126,6 +126,8 @@ describe('Test it can get all favorites for a playlist with playlists_favorites
 describe('Test the delete playlist endpoint', () => {
   beforeEach(async () => {
     await database.raw('truncate table playlists cascade');
+    await database.raw('truncate table playlists_favorites cascade');
```
Comment (lrs8810) [inline tests/playlistFavorites.spec.js]: I added these truncate/cascades to help me get past the foreign key/duplicate errors we were experiencing earlier

File: routes/api/v1/playlists.js (hunk 1 above)
Comment (zacisaacson) [reply routes/api/v1/playlists.js]: I think it looks great!

File: tests/playlistFavorites.spec.js (hunk 2 above)
Comment (zacisaacson) [reply tests/playlistFavorites.spec.js]: Good call, I had to add that to a couple tests as well

**Pull request merged**